qtaskbarcontrol.cpp

```diff
@@ -69,53 +72,53 @@ void QTaskbarControl::setWindowsBadgeTextColor(const QColor &color)
 
 void QTaskbarControl::setProgressVisible(bool progressVisible)
 {
-	if (d->progressVisible == progressVisible)
+	if (d->_progressVisible == progressVisible)
 		return;
 
-	d->progressVisible = progressVisible;
-	d->setProgress(progressVisible, d->progress);
+	d->_progressVisible = progressVisible;
+	d->setProgress(progressVisible, d->_progress);
 	emit progressVisibleChanged(progressVisible);
 }
 
 void QTaskbarControl::setProgress(double progress)
 {
-	if (qFuzzyCompare(d->progress, progress))
+	if (qFuzzyCompare(d->_progress, progress))
 		return;
 
-	d->progress = progress;
-	d->setProgress(d->progressVisible, progress);
+	d->_progress = progress;
+	d->setProgress(d->_progressVisible, progress);
 	emit progressChanged(progress);
 }
 
 void QTaskbarControl::setCounterVisible(bool counterVisible)
 {
-	if (d->counterVisible == counterVisible)
+	if (d->_counterVisible == counterVisible)
 		return;
 
-	d->counterVisible = counterVisible;
-	d->setCounter(counterVisible, d->counter);
+	d->_counterVisible = counterVisible;
+	d->setCounter(counterVisible, d->_counter);
 	emit counterVisibleChanged(counterVisible);
 }
 
 void QTaskbarControl::setCounter(int counter)
 {
-	if (d->counter == counter)
+	if (d->_counter == counter)
 		return;
 
-	d->counter = counter;
-	d->setCounter(d->counterVisible, counter);
+	d->_counter = counter;
+	d->setCounter(d->_counterVisible, counter);
 	emit counterChanged(counter);
 }
 
 bool QTaskbarControl::eventFilter(QObject *watched, QEvent *event)
 {
-	if(watched == parent()) {
-		if(event->type() == QEvent::Show) {
-			auto wid = qobject_cast<QWidget*>(parent());
-			if(wid)
-				d->setWindow(wid->windowHandle());
+	if (event->type() == QEvent::Show) {
+		auto widget = qobject_cast<QWidget*>(watched);
+		if (widget) {
+			d->setWindow(widget->windowHandle());
+			widget->removeEventFilter(this);
```

Comment:
Please do not remove the event filter here, as theoretically, it is possible for the widget to get a new window assigned, if it is reparented.

Reply:
I tested with the [repaint](https://doc.qt.io/qt-5/qwidget.html#repaint) and it works. I think that `QWindow` assigned only once. Even if `QWindow` changes somehow, show event will not be called, so this event filter will not help. I would suggest letting the user deal with this.

Reply:
I tested it with [QWindow::setParent](https://doc.qt.io/qt-5/qwindow.html#setParent) and with [QWidget::setParent](https://doc.qt.io/qt-5/qwidget.html#setParent) - `QWindow` for every widget stills the same. **It not works even with the event filter** if I change parent of the window with taskbar progress to another because there is no window to display taskbar progress (taskbar progress will disappear). But if I remove parent back, the window with the progress will appear again. I tested it too.

Reply:
In that case, keep it as you changed it.